### PR TITLE
Fix react-intl example

### PR DESCRIPTION
**Pull request by istarkov**
react-intl does not work otherwise

### CODE REVIEW

**Comment by timneutkens** [comment]
I've merged this change around 6 times so far 😅Someone coming along bringing what it is now, then someone else says it doesn't work and changes it back. Please make 110% sure this is the correct way.

File: examples/with-react-intl/pages/_app.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ export default class MyApp extends App {
     // Get the `locale` and `messages` from the request object on the server.
     // In the browser, use the same values that the server serialized.
     const { req } = ctx
-    const { locale, messages } = req || window.__NEXT_DATA__.props.pageProps
+    const { locale, messages } = req || window.__NEXT_DATA__.props
```
Comment (timneutkens) [inline examples/with-react-intl/pages/_app.js]: Also, this shouldn't use `__NEXT_DATA__` it should use `pageProps`

Comment (istarkov) [reply examples/with-react-intl/pages/_app.js]: There are no `pageProps` at client in this method.
As I see the only way to get something in `_app.js` added by server into `req` object is to return this in his own `getInitialProps` and then read from `__NEXT_DATA__`

**Comment by istarkov** [comment]
>I've merged this change around 6 times so far 😅Someone coming along bringing what it is now, then someone else says it doesn't work and changes it back. Please make 110% sure this is the correct way.

I think you are mistaken here ;-) it seems like before `_app` Higher Order Component was used for intl logic and in that case reading from `pageProps` was right solution, as default app put page data into `pageProps` [see here](https://github.com/istarkov/next.js/blob/canary/lib/app.js#L12)

But here `_app.js` is used itself and that data is in [__NEXT_DATA__.props](https://github.com/istarkov/next.js/blob/canary/server/render.js#L156)

**Review by timneutkens — APPROVED** [review]
Lets just merge this then.

**Pull request merged**